src/main/MQ2Commands.cpp

```diff
@@ -2670,41 +2656,28 @@ void DoAbility(SPAWNINFO* pChar, char* szLine)
 		return;
 	}
 
-	int abilityNum = GetIntFromString(szBuffer, 0);
-	if (abilityNum > 0)
-	{
-		// Check if user wants us to activate an ability by its "real id" (?)
-		if (abilityNum > 6 && abilityNum < NUM_SKILLS)
-		{
-			int token = pSkillMgr->GetNameToken(abilityNum);
-			if (token != 0)
-			{
-				strcpy_s(szBuffer, pStringTable->getString(token));
-			}
-		}
-		else
-		{
-			// passthrough to original function
-			cmdDoAbility(pChar, szLine);
-			return;
-		}
-	}
-
 	// scan for matching abilities name
-	for (int Index = 0; Index < 128; Index++)
+	for (int Index = 0; Index < NUM_SKILLS + NUM_INNATE; ++Index)
 	{
-		if (Index < NUM_SKILLS && pSkillMgr->pSkill[Index]->Activated
-			|| Index > NUM_SKILLS && pProfile->InnateSkill[Index - 100] != 0xFF)
+		if (const char* skill_name = GetSkillName(Index))
 		{
-			if (!_stricmp(szBuffer, szSkills[Index]))
+			if (ci_equals(szBuffer, skill_name))
 			{
-				if (!HasSkill(Index))
+				if (HasSkillOrInnate(Index))
 				{
-					WriteChatf("you do not have this skill (%s)", szBuffer);
-					return;
+					if (SkillOrInnateIsActivatable(Index))
+					{
+						pLocalPC->UseSkill(static_cast<uint8_t>(Index), pLocalPlayer);
+					}
+					else
+					{
+						WriteChatf("DoAbility: Skill is not activatable (%s)", szBuffer);
```

Comment:
This message is not consistent with the others ("DoAbility:")

Reply:
If I make the others consistent, I potentially break any events.  I can make this one consistent with the others but there's no indication where the message came from.

Reply:
That seems like its already a problem for the rest of this function, then? Its no worse off than it already is.

Reply:
Six of one, but I prefer the inconsistency and clarity on the updated function over the consistency without clarity.

Reply:
Well then it should say /doability then, right? DoAbility isn't meaningful to anybody but us

Reply:
Fair